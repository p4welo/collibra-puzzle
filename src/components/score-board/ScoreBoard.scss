@import "../../styles/colors";
@import "../../styles/variables";

.ScoreBoard {
  width: 300px;
}

.ScoreBoard__header {
  margin-bottom: 10px;
  text-align: center;
}

.ScoreBoard__table {
  background: $light;
  border: $borderWidth * 2 solid $blue;
  text-align: right;
  width: 100%;

  td, th {
    padding: 12px 20px;
  }
}
