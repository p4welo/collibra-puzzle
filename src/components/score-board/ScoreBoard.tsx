import React, { PureComponent } from 'react';
import './ScoreBoard.scss';

interface ScoreBoardProps {
  scores: number[];
}

export class ScoreBoard extends PureComponent<ScoreBoardProps> {

  render() {
    const { scores = [] } = this.props;
    return (
        <div className='ScoreBoard'>
          <h3 className='ScoreBoard__header'>Scoreboard:</h3>

          <table className='ScoreBoard__table'>
            <thead>
            <tr>
              <th>Attempt No</th>
              <th>Time</th>
            </tr>
            </thead>

            <tbody>
            {
              scores.length > 0 ?
                  scores
                      .map((score: number, i: number) =>
                          <tr key={i} data-testid='score-row'>
                            <td>{i + 1}</td>
                            <td>{score} ms</td>
                          </tr>
                      ) :

                  <tr>
                    <td colSpan={2}>No results yet.</td>
                  </tr>
            }
            </tbody>
          </table>
        </div>
    );
  }
}
