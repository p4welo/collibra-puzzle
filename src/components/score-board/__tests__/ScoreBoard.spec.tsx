import React from 'react';
import { render } from '@testing-library/react';
import { ScoreBoard } from '../ScoreBoard';

const scoreListMock: number[] = [
  10000,
  9000,
  8000,
  7000,
  11000
];

describe('ScoreBoard component', () => {
  it('should display all scores', () => {
    const { getAllByTestId } = render(<ScoreBoard scores={scoreListMock}/>);
    expect(getAllByTestId('score-row').length).toEqual(scoreListMock.length);
  });

  it('should display "No results" text when empty score list given', () => {
    const { queryByText } = render(<ScoreBoard scores={[]}/>);
    expect(queryByText('No results yet.')).toBeInTheDocument();
  });

  it('should not display "No results" text when filled score list given', () => {
    const { queryByText } = render(<ScoreBoard scores={scoreListMock}/>);
    expect(queryByText('No results yet.')).toBeNull();
  });

  it('should match snapshot', () => {
    const { container } = render(<ScoreBoard scores={scoreListMock}/>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
