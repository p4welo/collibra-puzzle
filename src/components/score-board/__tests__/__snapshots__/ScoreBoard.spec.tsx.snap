// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard component should match snapshot 1`] = `
<div
  class="ScoreBoard"
>
  <h3
    class="ScoreBoard__header"
  >
    Scoreboard:
  </h3>
  <table
    class="ScoreBoard__table"
  >
    <thead>
      <tr>
        <th>
          Attempt No
        </th>
        <th>
          Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        data-testid="score-row"
      >
        <td>
          1
        </td>
        <td>
          10000
           ms
        </td>
      </tr>
      <tr
        data-testid="score-row"
      >
        <td>
          2
        </td>
        <td>
          9000
           ms
        </td>
      </tr>
      <tr
        data-testid="score-row"
      >
        <td>
          3
        </td>
        <td>
          8000
           ms
        </td>
      </tr>
      <tr
        data-testid="score-row"
      >
        <td>
          4
        </td>
        <td>
          7000
           ms
        </td>
      </tr>
      <tr
        data-testid="score-row"
      >
        <td>
          5
        </td>
        <td>
          11000
           ms
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
